## CARS EDA ###

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
from matplotlib import pyplot as plt

### Executive Summary  :

The period between 1970 and 1982 marked a significant shift in the United States car industry. American production shifted from heavy, powerful six and eight-cylinder cars with poor gas mileage to lighter, less powerful, four-cylinder cars with higher fuel efficiency.

The global auto industry, including Americans and their European and Japanese competitors, raised overall miles per gallon (MPG) by focusing on four-cylinder cars and making them more fuel-efficient.

In [8]:
cars_country=pd.read_csv(r'C:\Users\goyal\OneDrive\Documents\CAR EDA\cars_country.csv',encoding='utf-8')
cars_multi=pd.read_csv(r'C:\Users\goyal\OneDrive\Documents\CAR EDA\cars_multi.csv',encoding='utf-8')
cars_price=pd.read_csv(r'C:\Users\goyal\OneDrive\Documents\CAR EDA\cars_price.csv',encoding='utf-8')

In [9]:
cars_country.head()

origin country
0       1     USA
1       2  Europe
2       3   Japan

In [10]:
cars_multi.head()

id   mpg  cylinders  displacement  horsepower  weight  acceleration  model   
0   1  18.0          8         307.0       130.0    3504          12.0     70  \
1   2  15.0          8         350.0       165.0    3693          11.5     70   
2   3  18.0          8         318.0       150.0    3436          11.0     70   
3   4  16.0          8         304.0       150.0    3433          12.0     70   
4   5  17.0          8         302.0       140.0    3449          10.5     70   

   origin                   car_name  
0       1  chevrolet chevelle malibu  
1       1          buick skylark 320  
2       1         plymouth satellite  
3       1              amc rebel sst  
4       1                ford torino

In [13]:
cars_price.head()

id        price
0   1  25561.59078
1   2  24221.42273
2   3  27240.84373
3   4  33684.96888
4   5  20000.00000

Here we are 3 datasets, we need to join these datasets to form a single datasets.

In [18]:
semi_data=cars_multi.merge(cars_price,how='left',left_on='id', right_on='id',suffixes=('_left', '_right'))
final_data=semi_data.merge(cars_country,how='left',left_on='origin',right_on='origin',suffixes=('_left', '_right'))

In [19]:
final_data.head()

id   mpg  cylinders  displacement  horsepower  weight  acceleration  model   
0   1  18.0          8         307.0       130.0    3504          12.0     70  \
1   2  15.0          8         350.0       165.0    3693          11.5     70   
2   3  18.0          8         318.0       150.0    3436          11.0     70   
3   4  16.0          8         304.0       150.0    3433          12.0     70   
4   5  17.0          8         302.0       140.0    3449          10.5     70   

   origin                   car_name        price country  
0       1  chevrolet chevelle malibu  25561.59078     USA  
1       1          buick skylark 320  24221.42273     USA  
2       1         plymouth satellite  27240.84373     USA  
3       1              amc rebel sst  33684.96888     USA  
4       1                ford torino  20000.00000     USA

In [22]:
final_data=final_data.drop('origin',axis=1)

In [23]:
final_data.head()

id   mpg  cylinders  displacement  horsepower  weight  acceleration  model   
0   1  18.0          8         307.0       130.0    3504          12.0     70  \
1   2  15.0          8         350.0       165.0    3693          11.5     70   
2   3  18.0          8         318.0       150.0    3436          11.0     70   
3   4  16.0          8         304.0       150.0    3433          12.0     70   
4   5  17.0          8         302.0       140.0    3449          10.5     70   

                    car_name        price country  
0  chevrolet chevelle malibu  25561.59078     USA  
1          buick skylark 320  24221.42273     USA  
2         plymouth satellite  27240.84373     USA  
3              amc rebel sst  33684.96888     USA  
4                ford torino  20000.00000     USA

In [25]:
final_data.shape

(398, 11)

In [26]:
final_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 398 entries, 0 to 397
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   id            398 non-null    int64  
 1   mpg           398 non-null    float64
 2   cylinders     398 non-null    int64  
 3   displacement  398 non-null    float64
 4   horsepower    392 non-null    float64
 5   weight        398 non-null    int64  
 6   acceleration  398 non-null    float64
 7   model         398 non-null    int64  
 8   car_name      398 non-null    object 
 9   price         398 non-null    float64
 10  country       398 non-null    object 
dtypes: float64(5), int64(4), object(2)
memory usage: 34.3+ KB


In [27]:
final_data.describe()

id         mpg   cylinders  displacement  horsepower   
count  398.000000  398.000000  398.000000    398.000000  392.000000  \
mean   199.500000   23.514573    5.454774    193.425879  104.469388   
std    115.036951    7.815984    1.701004    104.269838   38.491160   
min      1.000000    9.000000    3.000000     68.000000   46.000000   
25%    100.250000   17.500000    4.000000    104.250000   75.000000   
50%    199.500000   23.000000    4.000000    148.500000   93.500000   
75%    298.750000   29.000000    8.000000    262.000000  126.000000   
max    398.000000   46.600000    8.000000    455.000000  230.000000   

            weight  acceleration       model         price  
count   398.000000    398.000000  398.000000    398.000000  
mean   2970.424623     15.568090   76.010050  29684.473444  
std     846.841774      2.757689    3.697627   9832.263303  
min    1613.000000      8.000000   70.000000   1598.073370  
25%    2223.750000     13.825000   73.000000  23109.802772  
50%    2803.500000     15.500000   76.000000  30000.000000  
75%    3608.000000     17.175000   79.000000  36429.918887  
max    5140.000000     24.800000   82.000000  53745.938060

### Getting some head insights from the datasets

#### In the 'final_data' dataset we observe that-:

1. Minimum mpg(Milleage Per Gallon) of the cars are observed to be 9.0 and Maximum mpg of the cars are observed to be 46.6 across all the countries(USA,JAPAN,EUROPE).

2. Cars with the minimum cylinders we have is 3 and cars with the maximum cylinders we have 8.
3. Cars with the minimum displacement we have 68 and cars with the maximum displacement we have 455.
4. Cars with the minimum horsepower we have 46 and cars with the maximum horsepower we have 230.
5. Cars with the minimum weight we have 1613 and cars with the maximum weight we have 5140.
6. Cars with the minimum acceleration we have 8 and cars with the maximum acceleration we have 24.8.
7. Cars with the lowest model we have 70 and cars with the latest model we have 82.
8. Cars with the lowest price we have 1598.07 and cars with the latest model we have 53745.